filters/include/pcl/filters/experimental/voxel_grid_label.h

```diff
@@ -0,0 +1,153 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/filters/experimental/voxel_grid.h>
+#include <pcl/point_types.h>
+
+#include <map>
+
+namespace pcl {
+namespace experimental {
+
+struct LabeledVoxel {
+  union Centroid {
+    Centroid() {}
+    ~Centroid() {}
+
+    CentroidPoint<PointXYZRGBL> all_fields;
+    Eigen::Array4f xyz;
+  };
```

Comment:
Maybe this can be templated on PointT for CentroidPoint<PointT>?

Basically, a struct that encapsulates the logic of downsample, union and num_pt.

The logic for add, get can be modified as needed by inheriting without virtual inheritance.

Reply:
Done, `LabeledVoxel` is inheriting from `Voxel` from `VoxelGrid`